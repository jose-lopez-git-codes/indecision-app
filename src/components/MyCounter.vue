<template name="component-name">
  <section>
    <h3>Counter: {{ counter }}</h3>
    <h3 data-testid="square-label">Square: {{ squareCounter }}</h3>
    <div>
      <button class="btn-increment" @click="counter++">+1</button>
      <button class="btn-decrement" @click="counter--">-1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter';

interface Props {
  value: number;
}

const props = defineProps<Props>();
const { counter, squareCounter } = useCounter(props.value);
</script>

<style scoped>
@reference 'tailwindcss';
.btn-increment {
  @apply ml-2 p-2 bg-blue-500 rounded hover:bg-blue-700;
}
.btn-decrement {
  @apply ml-2 p-2 bg-red-500 rounded mr-2 hover:bg-red-700;
}
</style>
