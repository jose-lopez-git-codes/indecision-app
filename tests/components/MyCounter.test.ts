import { describe, expect, test } from 'vitest';
import { mount } from '@vue/test-utils';
import MyCounter from '@/components/MyCounter.vue';

describe('<MyCounter />', () => {
  test('should match snaptshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 10,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value,
      },
    });
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );

    const [counterLabel, squareLabel] = wrapper.findAll('h3');
    expect(counterLabel.text()).toContain(`Counter: ${value}`);
    expect(squareLabel.text()).toContain(`Square: ${value * value}`);
  });

  test('increments counter when +1 button is clicked ', async () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value,
      },
    });

    const btnIncrement = wrapper.find('button.btn-increment');

    await btnIncrement.trigger('click');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value + 1}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${(value + 1) * (value + 1)}`,
    );
  });

  test('decrements counter when -1 button is clicked twice', async () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value,
      },
    });

    const btnDecrement = wrapper.find('button.btn-decrement');
    await btnDecrement.trigger('click');
    await btnDecrement.trigger('click');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value - 2}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${(value - 2) * (value - 2)}`,
    );
  });
});
