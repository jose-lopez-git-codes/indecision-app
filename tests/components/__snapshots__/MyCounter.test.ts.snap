// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should match snaptshot 1`] = `
"<section data-v-0352012b="">
  <h3 data-v-0352012b="">Counter: 10</h3>
  <h3 data-v-0352012b="" data-testid="square-label">Square: 100</h3>
  <div data-v-0352012b=""><button data-v-0352012b="" class="btn-increment">+1</button><button data-v-0352012b="" class="btn-decrement">-1</button></div>
</section>"
`;
